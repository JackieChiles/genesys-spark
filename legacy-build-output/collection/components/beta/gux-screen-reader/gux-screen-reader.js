import { h } from '@stencil/core';
import { trackComponent } from '@utils/tracking/usage';
/**
 * @slot - text
 */
export class GuxScreenReader {
  componentWillLoad() {
    trackComponent(this.root);
  }
  render() {
    return (h("span", { class: "gux-sr-only" }, h("slot", null)));
  }
  static get is() { return "gux-screen-reader-beta"; }
  static get encapsulation() { return "shadow"; }
  static get originalStyleUrls() {
    return {
      "$": ["gux-screen-reader.less"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["gux-screen-reader.css"]
    };
  }
  static get elementRef() { return "root"; }
}
