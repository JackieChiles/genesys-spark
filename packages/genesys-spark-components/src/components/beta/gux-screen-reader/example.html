<h1>gux-screen-reader-beta</h1>

<div>
  There is visible but
  <gux-screen-reader-beta>this is not</gux-screen-reader-beta>
</div>
