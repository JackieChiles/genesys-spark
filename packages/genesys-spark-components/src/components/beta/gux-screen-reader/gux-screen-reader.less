@import (reference) '../../../style/typography.less';

.gux-sr-only {
  .gux-sr-only-clip();
}
