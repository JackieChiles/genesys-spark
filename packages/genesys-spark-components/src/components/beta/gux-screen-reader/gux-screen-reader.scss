@use '~genesys-spark/dist/scss/mixins.scss';

.gux-sr-only {
  @include mixins.gux-sr-only-clip;
}
