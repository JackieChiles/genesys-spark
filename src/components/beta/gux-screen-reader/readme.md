# gux-date-beta



<!-- Auto Generated Below -->


## Slots

| Slot | Description |
| ---- | ----------- |
|      | text        |


## Dependencies

### Used by

 - [gux-form-field-time-picker](../../stable/gux-form-field/components/gux-form-field-time-picker)

### Graph
```mermaid
graph TD;
  gux-form-field-time-picker --> gux-screen-reader-beta
  style gux-screen-reader-beta fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
