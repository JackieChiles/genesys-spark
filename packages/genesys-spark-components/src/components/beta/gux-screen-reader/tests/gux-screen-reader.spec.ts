import { newSpecPage } from '@test/specTestUtils';
import { GuxScreenReader } from '../gux-screen-reader';

const components = [GuxScreenReader];
const language = 'en';

describe('gux-screen-reader-beta', () => {
  describe('#render', () => {
    it('should render component as expected', async () => {
      const html = `
        <gux-screen-reader-beta>
          This is content
        </gux-screen-reader-beta>
      `;
      const page = await newSpecPage({ components, html, language });

      expect(page.root).toMatchSnapshot();
    });
  });
});
