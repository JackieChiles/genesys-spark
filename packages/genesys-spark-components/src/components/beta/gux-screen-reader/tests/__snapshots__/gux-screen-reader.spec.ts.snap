// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-screen-reader-beta #render should render component as expected 1`] = `
<gux-screen-reader-beta>
  <template shadowrootmode="open">
    <span class="gux-sr-only">
      <slot></slot>
    </span>
  </template>
  This is content
</gux-screen-reader-beta>
`;
